### PR TITLE
[ONNXIFI]Make TEST_P be able to show the test case name directly

### PR DESCRIPTION
**Pull request by zrphercule**
Before:
[ RUN      ] ONNXCppAllTest/ONNXCppDriverTest.ONNXCppDriverUnitTest/1
[       OK ] ONNXCppAllTest/ONNXCppDriverTest.ONNXCppDriverUnitTest/1
[ RUN      ] ONNXCppAllTest/ONNXCppDriverTest.ONNXCppDriverUnitTest/2
[       OK ] ONNXCppAllTest/ONNXCppDriverTest.ONNXCppDriverUnitTest/2

After:
[ RUN      ] ONNXCppAllTest/ONNXCppDriverTest.ONNXCppDriverUnitTest/test_maxpool_2d_strides
[       OK ] ONNXCppAllTest/ONNXCppDriverTest.ONNXCppDriverUnitTest/test_maxpool_2d_strides 
[ RUN      ] ONNXCppAllTest/ONNXCppDriverTest.ONNXCppDriverUnitTest/test_conv_with_strides_padding
[       OK ] ONNXCppAllTest/ONNXCppDriverTest.ONNXCppDriverUnitTest/test_conv_with_strides_padding

### CODE REVIEW

**Comment by zrphercule** [comment]
@houseroad yes we can, I can have another string store in ResolvedTestCase as its path. But so far the content is same thing, that is the path is the model graph name......

**Comment by zrphercule** [comment]
But I guess you are right, if the model is deprecated, or it has some special name for graph, then it doesnt make sense. let me add it.

**Comment by houseroad** [comment]
Maybe a helper function to extract the test name?

**Comment by zrphercule** [comment]
@houseroad The test nam can be the model.graph.name, also can be the directory name where we load the model. I am going to use the second one as test name now (Although these two names are same in our test data), and I just need to let ResolvedTestCase store where its data loaded from.

**Comment by houseroad** [comment]
@zrphercule we have two reasons to use folder name:
1) easier to parse the string comparing to protobuf
2) name can be empty, but folder won't be empty, so it's safer

**Comment by houseroad** [comment]
BTW, let's add the reasons in the comments :-)

File: onnx/backend/test/cpp/driver/test_driver.cc [hunk 1]
```diff
@@ -97,10 +100,10 @@ bool TestDriver::FetchAllTestCases(const std::string& target) {
   } else {
     try {
       do {
-        std::string entry_dname = file.name;
+        std::string entry_dname = file.name, full_entry_dname;
```
[the remark this answers is not in the document]
Comment (zrphercule) [reply onnx/backend/test/cpp/driver/test_driver.cc]: sure...

**Pull request merged**